### PR TITLE
NIFI-8329 - Updated dependencies that had no build/test failures when…

### PR DESCRIPTION
**Pull request by thenatog**
… updating.

NIFI-8329 - Not updating storm dependency in this change, as it is being removed in a separate ticket.

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by exceptionfactory — COMMENTED** [review]
Thanks for the work on this @thenatog!  Verified successful build and basic runtime behavior.  Reviewed change logs for Jackson and several other dependencies, no concerning issues noted.  See the one comment on the Jackson version of `nifi-azure`, otherwise looks good.

File: nifi-nar-bundles/nifi-azure-bundle/pom.xml [hunk 1]
```diff
@@ -28,7 +28,7 @@
     <properties>
         <azure-storage.version>8.4.0</azure-storage.version>
         <azure.core.version>1.6.0</azure.core.version>
-        <jackson.version>2.10.3</jackson.version>
+        <jackson.version>2.12.2</jackson.version>
```
[the remark this answers is not in the document]
Comment (thenatog) [reply nifi-nar-bundles/nifi-azure-bundle/pom.xml]: I thought I had issues with the property being picked up from this module, but I tested it again and it seems fine. Removed this with a new commit.

[the remark this answers is not in the document]
Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-azure-bundle/pom.xml]: Thanks for confirming and making the change.

**Review by exceptionfactory — APPROVED** [review]
Thanks again @thenatog! +1 Merging.

**Pull request closed without merging**